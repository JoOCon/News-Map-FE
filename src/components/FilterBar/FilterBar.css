.suggestion:hover {
  background-color: #b3b3b3;
  color: #fafafa; }

.location-input {
  font-size: 1.4rem;
  width: 12rem;
  position: absolute;
  z-index: 100;
  top: 2.9%;
  left: 5.5%;
  margin: 0 0 0 -6rem; }

.suggestion-list {
  color: #856c4b;
  width: 12.4rem;
  cursor: pointer;
  position: absolute;
  top: 5.5%;
  left: 5.6%;
  background-color: white;
  z-index: 100;
  margin: 0 0 0 -6.2rem; }

.suggestion {
  border: 0.8px solid gray;
  padding: 0.3rem;
  margin: 0; }

.active-suggestion {
  border: 0.8px solid gray;
  padding: 0.3rem;
  margin: 0;
  background-color: #b3b3b3; }
