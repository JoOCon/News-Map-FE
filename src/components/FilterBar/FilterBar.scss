.suggestion:hover {
  background-color: rgb(179, 179, 179);
  color: rgb(250, 250, 250);
}

.location-input {
  font-size: 1.4rem;
  width: 12rem;
  position: absolute;
  z-index: 100;
  top: 2.9%;
  left: 5.5%;
  margin: 0 0 0 -6rem;
}

.suggestion-list {
  color: rgb(133, 108, 75);
  width: 12.4rem;
  cursor: pointer;
  position: absolute;
  top: 5.5%;
  left: 5.6%;
  background-color: rgb(255, 255, 255);
  z-index: 100;
  margin: 0 0 0 -6.2rem;
}

.suggestion {
  border: 0.8px solid gray;
  padding: 0.3rem;
  margin: 0;
}

.active-suggestion {
  border: 0.8px solid gray;
  padding: 0.3rem;
  margin: 0;
  background-color: rgb(179, 179, 179);
}

// form {
//   position: absolute
//   left: 50%;
//   top: 0
// }
