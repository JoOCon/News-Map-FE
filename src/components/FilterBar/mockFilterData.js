export const mocksuggestions = [
  {
    city: 'East Las Vegas',
    city_ascii: 'East Las Vegas',
    county_fips: '32003',
    county_name: 'Clark',
    density: '2912.1',
    id: '1840033774',
    incorporated: 'False',
    lat: '36.0944',
    lng: '-115.0419',
    population: '',
    population_proper: '',
    source: 'point',
    state_id: 'NV',
    state_name: 'Nevada',
    timezone: 'America/Los_Angeles',
    zips: '89122'
  }
];
