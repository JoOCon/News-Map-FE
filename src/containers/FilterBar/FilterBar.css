.suggestion:hover {
  background-color: #b3b3b3;
  color: #fafafa; }

.location-input {
  font-size: 1.4rem;
  width: 12rem;
  position: absolute;
  z-index: 100;
  top: 2.9%;
  left: 2.5%;
  border-radius: 0.5rem;
  display: flex;
  justify-content: start;
  align-items: start;
  border: 0.5px solid #765959;
  box-shadow: 1.5px 1.5px #a0a0a0;
  background-color: #dde7fa;
  padding: 0.5rem; }

.location-input:hover {
  box-shadow: 2.5px 2.5px #a0a0a0;
  background-color: #ffffff; }

input:focus {
  outline-width: 0; }

.suggestion-list {
  color: #856c4b;
  width: 13.2rem;
  cursor: pointer;
  position: absolute;
  top: 6.7%;
  left: 2.5%;
  background-color: white;
  z-index: 100; }

.suggestion {
  border: 0.5px solid gray;
  padding: 0.3rem;
  margin: 0; }

.active-suggestion {
  border: 0.8px solid gray;
  padding: 0.3rem;
  margin: 0;
  background-color: #b3b3b3; }

@media only screen and (max-width: 600px) {
  .location-input {
    font-size: 1rem;
    width: 10rem; }
  .suggestion-list {
    width: 11rem; } }
