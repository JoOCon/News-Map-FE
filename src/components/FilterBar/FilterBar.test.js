import React from 'react';

import { FilterBar } from './FilterBar';
import * as suggest from './mockFilterData';

describe('FilterBar', () => {
  let wrapper;
  let mockFunction;

  beforeEach(() => {
    mockFunction = jest.fn();
    wrapper = shallow(
      <FilterBar setUserLocation={mockFunction} setEvents={mockFunction} />
    );
  });

  it('should matchSnapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should setState of location and suggestions when update value is called', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: { value: 'east las' }
    };
    wrapper.instance().updateValue(mockEvent);
    expect(wrapper.state().location).toEqual('east las');
    expect(wrapper.state().suggestions).toEqual(suggest.mocksuggestions);
    expect(wrapper.state().cursor).toEqual(0);
  });

  it('should return the proper string when escapeRegexCharacters is called with a string', () => {
    let result = wrapper.instance().escapeRegexCharacters('LAS VEGAS');
    expect(result).toEqual('LAS VEGAS');
  });

  it('should return an array of suggestions when getSuggestions is called', () => {
    const result = wrapper.instance().getSuggestions('east las');
    expect(result).toEqual(suggest.mocksuggestions);
  });

  it('should return the suggestion when getSuggestionValue is called', () => {
    const result = wrapper.instance().getSuggestionValue('east las vegas');
    expect(result).toEqual('east las vegas');
  });

  it('should decrement state of cursor when the event.keycode is 38', () => {
    const mockEvent = { keyCode: 38 };

    wrapper.setState({ cursor: 4 });
    wrapper.instance().handleKeyDown(mockEvent);

    expect(wrapper.state().cursor).toEqual(3);
  });

  it('should increment state of cursor when the event.keycode is 40', () => {
    const mockEvent = { keyCode: 40 };

    wrapper.setState({ cursor: 1, suggestions: [2, 3, 4] });
    wrapper.instance().handleKeyDown(mockEvent);

    expect(wrapper.state().cursor).toEqual(2);

    wrapper.setState({ cursor: 1, suggestions: [] });
    wrapper.instance().handleKeyDown(mockEvent);

    expect(wrapper.state().cursor).toEqual(1);
  });
});
