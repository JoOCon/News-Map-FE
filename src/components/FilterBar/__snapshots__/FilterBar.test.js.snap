// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar should matchSnapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterBar
    setEvents={[MockFunction]}
    setUserLocation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="location-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Search City"
          type="text"
          value=""
        />,
        <section
          className="suggestion-list"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "location-input",
          "onChange": [Function],
          "onKeyDown": [Function],
          "placeholder": "Search City",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "suggestion-list",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": Array [],
        "type": "section",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="location-input"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search City"
            type="text"
            value=""
          />,
          <section
            className="suggestion-list"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "location-input",
            "onChange": [Function],
            "onKeyDown": [Function],
            "placeholder": "Search City",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "suggestion-list",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": Array [],
          "type": "section",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
